Lib/ufoLib/__init__.py

```diff
@@ -121,11 +93,10 @@ class UFOReader(object):
 	"""Read the various components of the .ufo."""
 
 	def __init__(self, path):
-		if not os.path.exists(path):
-			raise UFOLibError("The specified UFO doesn't exist.")
-		self._path = path
+		self.fileSystem = FileSystem(path)
```

Comment:
I think it would be incredibly useful to optionally dependency inject an initiaized PyFilesystem2 implementation into UFOReader and UFOWriter.

Basically, to optionally use something different than OSFS or ZipFS. Here's a list of [PyFilesystem2 Builtin Filesystems](http://docs.pyfilesystem.org/en/latest/builtin.html) Why shouldn't we be able to chose any of these? Memory Filesystem sounds immediately useful to me (because it's fast) but something like an adapter to connect to a database would also be possible.

The implementation would probably need to happen in [class FileSystem](https://github.com/unified-font-object/ufoLib/blob/4d9291fe71725850fac1a0358b2188b640adc407/Lib/ufoLib/filesystem.py#L38)

Reply:
Right.  I'm mainly interested in Tar filesystem since it's fast and single-file.  A solution that only supports zip would be suboptimal.